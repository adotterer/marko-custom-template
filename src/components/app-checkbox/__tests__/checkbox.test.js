import { render } from "@marko/testing-library";
import userEvent from "@testing-library/user-event";
import Checkbox from "../index.marko";
import { axe } from "jest-axe";

describe("<app-checkbox>", () => {
  test("should toggle check when clicked", async () => {
    const user = userEvent.setup();
    const { container } = await render(Checkbox, {
      label: "Click here to subscribe",
    });

    const button = container.querySelector("button");
    await user.click(button);
    expect(button.classList.contains("checked")).toBe(true);
  });
  test("classList string should match snapshot", async () => {
    const user = userEvent.setup();
    const { container, getByText } = await render(Checkbox, {
      label: "Click here to subscribe",
    });

    const button = container.querySelector("button");
    await user.click(getByText(/click here to subscribe/i));
    expect(button.classList.value).toMatchSnapshot();
  });

  test("the checkbox is accessible", async () => {
    const { container } = await render(Checkbox, {
      label: "Click here to subscribe",
    });
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
