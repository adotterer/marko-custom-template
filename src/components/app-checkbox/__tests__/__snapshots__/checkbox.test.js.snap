// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<app-checkbox> classList string should match snapshot 1`] = `"app-checkbox checked app-button"`;
